### PR TITLE
Test for leapfrogging and ensure ordering is correct.

### PR DESCRIPTION
**Pull request by tgolsson**
This adds a test for leapfrogging and ensures the heap is always properly sorted.

### CODE REVIEW

File: crates/cervo-runtime/src/runtime.rs [hunk 1]
```diff
@@ -289,20 +289,50 @@ mod tests {
...
-        // queue should be 3, 0, 1, 2
+        // queue should be 3, 0, 1, 2.
+        // The below can run both 3 and 0 but only 3 has data.
         let res = runtime.run_for(Duration::from_secs_f32(0.07)).unwrap();
         assert_eq!(res.len(), 1);
         assert!(res.contains_key(&keys[3]));
 
         push(&mut runtime, &keys);
-        let res = runtime.run_for(Duration::from_secs_f32(0.161)).unwrap();
+        let res = runtime.run_for(Duration::from_secs_f32(0.165)).unwrap();
```
Comment (tgolsson) [inline crates/cervo-runtime/src/runtime.rs:299]: This flaked for me at 0.161

**Review by jaxs-ribs — APPROVED** [review]
Nice, LGTM!

File: crates/cervo-runtime/src/runtime.rs (hunk 1 above)
Comment (jaxs-ribs) [reply crates/cervo-runtime/src/runtime.rs:299]: I'm guessing overhead?

Comment (tgolsson) [reply crates/cervo-runtime/src/runtime.rs:299]: Yeah, guess so.

**Pull request merged**